### PR TITLE
Build Linuxbrew in Azure Pipelines and upload GitHub releases

### PR DESCRIPTION
**Pull request by mbautin**
- Automatically upload release archives as GitHub release artifacts once the build is done.
- Allow building in two modes: (1) with no restrictions and (2) avoiding SSE4 instructions, for older CPUs.
- Change Linuxbrew to Homebrew/Linuxbrew, and make things more general (not specific to Linuxbrew) in as many places as possible.

### CODE REVIEW

File: linuxbrew-common.sh [hunk 1]
```diff
@@ -69,3 +91,9 @@ get_escaped_sed_replacement_str() {
   local delim=${2:-/}
   sed "s/[$delim&\]/\\\\&/g" <<<$1
 }
+
+if [[ $OSTYPE == linux* ]]; then
+  YB_BREW_DIR_PREFIX=linuxbrew
+else
+  YB_BREW_DIR_PREFIX=homebrew
+fi
```
Comment (ttyusupov) [inline linuxbrew-common.sh:99]: Do we need to rename scripts to `brew-*` instead of `linuxbrew-*`?

File: linuxbrew-common.sh [hunk 2]
```diff
@@ -22,30 +22,52 @@ fi
...
+  local brew_dirname="$YB_BREW_DIR_PREFIX-$YB_BREW_TIMESTAMP"
+  local brew_path_prefix="$(realpath .)/$brew_dirname"
+  if [[ -n ${YB_BREW_SUFFIX:-} ]]; then
+    brew_path_prefix+="-$YB_BREW_SUFFIX"
+  fi
+  local len=${#brew_path_prefix}
   if [[ $len -gt $ABS_PATH_LIMIT ]]; then
-    echo "Linuxbrew link absolute path should be no more than $ABS_PATH_LIMIT bytes, but actual" \
-         "length is $len bytes: $brew_link" >&2
+    echo "Brew link absolute path should be no more than $ABS_PATH_LIMIT bytes, but actual" \
```
Comment (ttyusupov) [inline linuxbrew-common.sh]: Looks like it is not actually a link anymore?

File: linuxbrew-build.sh [hunk 3]
```diff
@@ -118,27 +111,19 @@ readonly LINUXBREW_PACKAGES=(
...
-readonly LINUXBREW_PACKAGES_DISABLED=(
-  maven
-)
-
 successful_packages=()
 failed_packages=()
 
 for package in "${LINUXBREW_PACKAGES[@]}"; do
-
-  if ( set -x; ./bin/brew install --build-from-source "$package" ); then
```
Comment (ttyusupov) [inline linuxbrew-build.sh:131]: Why we don't need `--build-from-source` anymore?

File: linuxbrew-common.sh [hunk 4]
```diff
@@ -22,30 +22,52 @@ fi
 
 declare -i -r ABS_PATH_LIMIT=85
 
-get_brew_link() {
```
Comment (ttyusupov) [inline linuxbrew-common.sh:25]: `linuxbrew-copy.sh` is also using `linuxbrew-common.sh`, so it needs to be updated as well.

File: README.md [hunk 5]
```diff
@@ -1,7 +1,5 @@
 # brew-build
 
-[![Build Status](https://dev.azure.com/yugabyte/yugabyte/_apis/build/status/YugaByte.brew-build?branchName=master)](https://dev.azure.com/yugabyte/yugabyte/_build/latest?definitionId=2&branchName=master)
-
 This set of wrappers scripts allows to build a customized version of Linuxbrew used by the YugaByte DB
 build process. We pre-install the Linuxbrew packages that we need, built with the appropriate compiler
 flags. We ensure that the full path to the root directory of the Linuxbrew installation is long enough so that we can
```
Comment (ttyusupov) [inline README.md]: General comment - we will need to test installing linuxbrew as a part of YB build and probably update "Using Linuxbrew and Homebrew" instruction.

**Review by ttyusupov — CHANGES_REQUESTED** [review]

File: linuxbrew-common.sh (hunk 2 above)
Comment (mbautin) [reply linuxbrew-common.sh]: Done.

File: linuxbrew-build.sh (hunk 3 above)
Comment (mbautin) [reply linuxbrew-build.sh:131]: Added back.

**Pull request merged**